src/App.tsx

```diff
@@ -1,12 +1,69 @@
 import React from 'react';
 import './App.scss';
+import { AllGoodsButton } from './components/AllGoodsButton';
+import { GoodsList } from './components/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { LoadFirst5Button } from './components/LoadFirst5Button';
+import { LoadRedGoodsButton } from './components/LoadRedGoodsButton';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[] | null;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: null,
+  };
+
+  getAllGoods = () => {
+    getAll()
+      .then(goods => {
+        this.setState({
+          goods,
+        });
+      });
+  };
+
+  loadFirst5Goods = () => {
+    get5First()
+      .then(goods => {
+        this.setState({
+          goods,
+        });
+      });
+  };
+
+  loadRedGoods = () => {
+    getRedGoods()
+      .then(goods => {
+        this.setState({
+          goods,
+        });
+      });
+  };
+
+  render() {
+    return (
+      <div className="App container col-6">
+        <h1>Dynamic list of Goods</h1>
+
+        <div className="d-flex justify-content-between">
+          <AllGoodsButton getGoods={this.getAllGoods} />
+
+          <LoadFirst5Button getGoods={this.loadFirst5Goods} />
+
+          <LoadRedGoodsButton getGoods={this.loadRedGoods} />
```

Comment:
why do you need 3 different components for buttons? or make 1 component Button and pass needed props there. Or just place all buttons here